### ISSUE TITLE
I can't run `flutter build` in github actions

### ISSUE THREAD

**Issue by baseplate-admin** [flutter/flutter#132834]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

```yaml
name: Build and test android APK

on:
    pull_request:
    push:
        branches:
            - main
jobs:
    build:
        runs-on: ubuntu-latest
        steps:
            - uses: actions/checkout@v3
            - uses: actions/setup-java@v2
              with:
                  distribution: 'zulu' # Same as my PC
                  java-version: '17'
            - uses: subosito/flutter-action@v2
              with:
                  flutter-version: '3.13.0-0.4.pre'
                  channel: 'beta'
            - run: flutter pub get
            - run: flutter build apk --release
```

```groovy
plugins {
    id "com.android.application"
    id "kotlin-android"
    id "dev.flutter.flutter-gradle-plugin"
}

ext{
    geckoviewChannel = "nightly"
    geckoviewVersion = "118.0.20230817214619"
}

def localProperties = new Properties()
def localPropertiesFile = rootProject.file('local.properties')
if (localPropertiesFile.exists()) {
    localPropertiesFile.withReader('UTF-8') { reader ->
        localProperties.load(reader)
    }
}

def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
if (flutterVersionCode == null) {
    flutterVersionCode = '1'
}

def flutterVersionName = localProperties.getProperty('flutter.versionName')
if (flutterVersionName == null) {
    flutterVersionName = '1.0'
}

android {
    namespace "moe.coreproject"
    compileSdkVersion flutter.compileSdkVersion
    ndkVersion flutter.ndkVersion

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_17
        targetCompatibility JavaVersion.VERSION_17
    }

    kotlinOptions {
        jvmTarget = '17'
    }

    sourceSets {
        main.java.srcDirs += 'src/main/kotlin'
    }

    defaultConfig {
        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
        applicationId "moe.coreproject"
        // You can update the following values to match your application needs.
        // For more information, see: https://docs.flutter.dev/deployment/android#reviewing-the-gradle-build-configuration.
        minSdkVersion  flutter.minSdkVersion
        targetSdkVersion flutter.targetSdkVersion
        versionCode flutterVersionCode.toInteger()
        versionName flutterVersionName
    }

    buildTypes {
        release {
            // TODO: Add your own signing config for the release build.
            // Signing with the debug keys for now, so `flutter run --release` works.
            signingConfig signingConfigs.debug
        }
    }
}

flutter {
    source '../..'
}

dependencies {
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
    implementation "org.mozilla.geckoview:geckoview-${geckoviewChannel}:${geckoviewVersion}"

}
```

### Actual results

App builds on local but doesn't build on github actions

### Logs

[2.txt](https://github.com/flutter/flutter/files/12377434/2.txt)
https://github.com/baseplate-admin/coreproject-android/actions/runs/5900618651

### Flutter Doctor output

<details>
<summary>Doctor output</summary>

```console

[√] Flutter (Channel beta, 3.13.0-0.4.pre, on Microsoft Windows [Version 10.0.22621.2134], locale en-US)
    • Flutter version 3.13.0-0.4.pre on channel beta at C:\tools\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e07cd41cb (9 days ago), 2023-08-08 21:20:03 -0700
    • Engine revision 73d89ca7c5
    • Dart version 3.1.0 (build 3.1.0-262.3.beta)
    • DevTools version 2.25.0

[√] Windows Version (Installed version of Windows is version 10 or higher)

[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\zarif\AppData\Local\Android\Sdk
    X cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[X] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[!] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.7.0)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.7.34003.232
    • Windows 10 SDK version 10.0.22000.0
    X Visual Studio is missing necessary components. Please re-run the Visual Studio installer for the "Desktop development with C++" workload, and include these components:
        MSVC v142 - VS 2019 C++ x64/x86 build tools
         - If there are multiple build tool versions available, install the latest
        C++ CMake tools for Windows
        Windows 10 SDK

[√] Android Studio (version 2022.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)

[√] VS Code (version 1.82.0-insider)
    • VS Code at C:\Users\zarif\AppData\Local\Programs\Microsoft VS Code Insiders
    • Flutter extension version 3.71.20230801

[√] Connected device (2 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.2134]
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 115.0.1901.203

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 3 categories.

```

</details>

**Comment by nightcap79** [comment]
Try to clean previous build by running:
`flutter clean`

**Comment by baseplate-admin** [comment]
> `flutter clean`

Do i have to clean on a fresh github action?

**Comment by baseplate-admin** [comment]
The issue was due to me using gradle `8.1`. 


The issue is fixed when i rolled back to `7.3`
